### PR TITLE
Add common tls config options that can be shared by both agent and orchestrator

### PR DESCRIPTION
**Pull request by soenkeliebau**
…

### CODE REVIEW

File: config/src/lib.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ use clap::{App, Arg, ArgMatches};
 
 // Include all "stolen" ripgrep code in this module
 mod ripgrep_config;
+pub mod tls_config;
```
Comment (lfrancke) [inline config/src/lib.rs:22]: Wouldn't it make sense to have this in a separate package?

So stackable-common/common-config
And have this as a pure library which can be reused independently?

File: config/src/tls_config.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
+use crate::ConfigDescription;
+use crate::ConfigOption;
+use crate::Configuration;
+
+/// The settings defined in this struct are common to all components that employ SSL
```
Comment (lfrancke) [inline config/src/tls_config.rs:5]: SSL -> TLS (just to go with the modern times...)

File: config/src/tls_config.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        takes_argument: true,
+        help: "The location of the truststore to use when connecting to the orchestrator.",
+        documentation:
+            "Specify a file in PKCS12 format that should be used to check if certificates\
+        are signed by a trusted authority. \
+        Any certificate that was signed with the private key belonging to one of the public keys\
+        in this truststore will be accepted as a valid certificte by this client.",
+    };
+
+    pub const TRUSTSTORE_PASSWORD: ConfigOption = ConfigOption {
```
Comment (lfrancke) [inline config/src/tls_config.rs:51]: Have you checked how TLS works in Rust? Are things called Key- & Truststore, do both concepts exist etc.?

File: config/src/tls_config.rs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    };
+
+    // TODO: Define sensible defaults
+    pub const ENABLED_PROTOCOLS: ConfigOption = ConfigOption {
+        name: "tls-enabled-protocols",
+        default: "",
+        required: false,
+        takes_argument: true,
+        help: "A list of acceptable protocol versions to use.",
+        documentation: "This defines the protocol versions that may be used. Any client trying to \
```
Comment (lfrancke) [inline config/src/tls_config.rs:81]: What happens if left blank?

**Comment by soenkeliebau** [comment]
This PR is not necessary any more due. When we moved to using K8S client libraries, we inherited all the SSL config stuff that they offer, so for now, no need to roll our own.
I'll close this for now.

**Pull request closed without merging**